CZ/pyranda_rayleigh_taylor/rayleigh_taylor.py

```diff
@@ -0,0 +1,646 @@
+import argparse
+import sys
+import textwrap
+import numpy
+import matplotlib.pyplot as plt
+from matplotlib import cm
+
+from pyranda import pyrandaSim, pyrandaBC, pyrandaTimestep
+
+
+# Setup some defaults
+DEFAULTS = {
+    "tstop": 100.0,
+    "stop-width-fraction": 0.5,
+    "max_iter": 10000
+}
+
+def run_sim(args):
+    Npts = args.num_points
+    dim = args.dim
+
+    if dim == "2D":
+        is2D = True
+    elif dim == "3D":
+        is2D = False
+    else:
+        raise ValueError(f"Unknown dim value '{dim}'")
+
+    if 'seed' in args:
+        random_ic_seed = args.seed
+    else:
+        random_ic_seed = None
+
+    headless = args.headless
+
+    ## Define a mesh
+    if is2D:
+        widthToHeight = 4.0
+        xwidth = widthToHeight*numpy.pi
+        problem = "RAYLEIGH_TAYLOR_2D"
+        imesh = (
+            """
+        xdom = (-xwidth/2.0, xwidth/2.0 , int(Npts*widthToHeight), periodic=False)
+        ydom = (0.0, 2*pi*FF,  Npts, periodic=True)
+        zdom = (0.0, 2*pi*FF,  1, periodic=True)
+        """.replace(
+                "Npts", str(Npts)
+            )
+            .replace("pi", str(numpy.pi))
+            .replace("FF", str(float(Npts - 1) / Npts))
+            .replace("xwidth", str(xwidth))
+            .replace("widthToHeight",str(widthToHeight))
+        )
+        waveLength = 4
+    else:
+        widthToHeight = 3.0
+        xwidth = widthToHeight*numpy.pi
+        problem = "RAYLEIGH_TAYLOR_3D"
+        imesh = (
+            """
+        xdom = (-xwidth/2.0, xwidth/2.0, int(Npts*widthToHeigth), periodic=False)
+        ydom = (0.0, 2*pi*FF,  Npts, periodic=True)
+        zdom = (0.0, 2*pi*FF,  Npts, periodic=True)
+        """.replace(
+                "Npts", str(Npts)
+            )
+            .replace("pi", str(numpy.pi))
+            .replace("FF", str(float(Npts - 1) / Npts))
+            .replace("xwidth", str(xwidth))
+        )
+        waveLength = 4
```

Comment:
Wondering if we should just change this to a single string for imesh that's located outside of the if/else since they're so similar. Something like:
```python
if is2D:
    widthToHeight = 4.0
    xwidth = widthToHeight*numpy.pi
    problem = "RAYLEIGH_TAYLOR_2D"
    waveLength = 4
    zpts = 1
else:
    widthToHeight = 3.0
    xwidth = widthToHeight*numpy.pi
    problem = "RAYLEIGH_TAYLOR_3D"
    waveLength = 4
    zpts = Npts

imesh = (
    """
    xdom = (-xwidth/2.0, xwidth/2.0, int(Npts*widthToHeight), periodic=False)
    ydom = (0.0, 2*pi*FF,  Npts, periodic=True)
    zdom = (0.0, 2*pi*FF,  zpts, periodic=True)
    """.replace(
            "Npts", str(Npts)
        )
        .replace("pi", str(numpy.pi))
        .replace("FF", str(float(Npts - 1) / Npts))
        .replace("xwidth", str(xwidth))
        .replace("widthToHeight",str(widthToHeight))
        .replace("zpts",str(zpts))
    )
```